araig_gui/src/araig_gui/araig_gui.cpp

```diff
@@ -0,0 +1,230 @@
+#include <thread>
+#include <chrono>
+#include <stdlib.h>
+#include <time.h>
+#include <sstream>
+
+AraigGui::AraigGui()
+  : rqt_gui_cpp::Plugin()
+  , widget_(0)
+{
+  setObjectName("AraigGui");
+}
+
+void AraigGui::initPlugin(qt_gui_cpp::PluginContext& context)
+{
+  QStringList argv = context.argv();
+  widget_ = new QWidget();
+
+  ui_.setupUi(widget_);
+  context.addWidget(widget_);
+  connect(ui_.pbTestStart, SIGNAL(clicked()), this, SLOT(on_pbTestStart_clicked()));
+  connect(ui_.pbTestStop, SIGNAL(clicked()), this, SLOT(on_pbTestStop_clicked()));
+  connect(ui_.pbTestReset, SIGNAL(clicked()), this, SLOT(on_pbTestReset_clicked()));
+  connect(ui_.pbTestSucc, SIGNAL(clicked()), this, SLOT(on_pbTestSucc_clicked()));
+  connect(ui_.pbTestFail, SIGNAL(clicked()), this, SLOT(on_pbTestFail_clicked()));
+
+  nh_ = getNodeHandle();
+  name_inputs_ = {"test_completed", "test_succeeded", "test_failed"};
+  name_outputs_ = {"start_test", "interrupt_test", "reset_test", "check_succeed", "check_failed"};
+  num_inputs_ = int(name_inputs_.size());
+  num_outputs_ = int(name_outputs_.size());
+  input_subs_.resize(static_cast<std::vector<ros::Publisher>::size_type>(num_inputs_));
+  input_states_.resize(static_cast<std::vector<bool>::size_type>(num_inputs_));
+  output_pubs_.resize(static_cast<std::vector<ros::Subscriber>::size_type>(num_outputs_));
+  output_states_.resize(static_cast<std::vector<bool>::size_type>(num_outputs_));
+  spawnPubs();
+  spawnSubs();
+  stateInit();
+  ROS_INFO_STREAM("[GUI]: ARAIG_TEST_GUI initialized!");
+
+}
+
+void AraigGui::shutdownPlugin()
+{
+}
+
+void AraigGui::saveSettings(qt_gui_cpp::Settings& plugin_settings,
+    qt_gui_cpp::Settings& instance_settings) const
+{
+}
+
+void AraigGui::restoreSettings(const qt_gui_cpp::Settings& plugin_settings,
+    const qt_gui_cpp::Settings& instance_settings)
+{
+}
+
+void AraigGui::spawnPubs()
+{
+  for(int i=0; i<num_outputs_; i++)
+  {
+    std::string topicName = "/signal/ui/" + name_outputs_[static_cast<std::vector<std::string>::size_type>(i)];
+    output_pubs_[static_cast<std::vector<ros::Publisher>::size_type>(i)]
+        = nh_.advertise<araig_msgs::BoolStamped>(topicName, 10);
+    ROS_INFO_STREAM("[GUI]: Spawned output topic publisher " << i << " for : " << topicName);
+  }
+}
+
+void AraigGui::spawnSubs()
+{
+  for(int i=0; i<num_inputs_; i++)
+  {
+    std::string topicName = "/signal/runner/" + name_inputs_[static_cast<std::vector<std::string>::size_type>(i)];
+    input_subs_[static_cast<std::vector<ros::Subscriber>::size_type>(i)]
+        = nh_.subscribe<araig_msgs::BoolStamped>(topicName, 10, boost::bind(&AraigGui::callbackBool, this, _1, i));
+    ROS_INFO_STREAM("[GUI]: Spawned input topic subscriber " << i << " for : " << topicName);
+  }
+}
+
+void AraigGui::callbackBool(const araig_msgs::BoolStamped::ConstPtr &msg, const int &topicIdx)
+{
+  input_states_[static_cast<std::vector<bool>::size_type>(topicIdx)] = msg->data? true:false;
```

Comment:
msg->data is already a bool, so you can assign that directly instead of checking it